servers/zms/src/main/java/com/yahoo/athenz/zms/notification/NotificationManager.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 Oath Holdings Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.yahoo.athenz.zms.notification;
+
+import com.yahoo.athenz.common.server.notification.Notification;
+import com.yahoo.athenz.common.server.notification.NotificationService;
+import com.yahoo.athenz.common.server.notification.NotificationServiceFactory;
+import com.yahoo.athenz.zms.*;
+import com.yahoo.athenz.zms.store.AthenzDomain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class NotificationManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NotificationManager.class);
+
+    NotificationServiceFactory notificationServiceFactory;
+    NotificationService notificationService;
+    ScheduledExecutorService scheduledExecutor;
+    private DBService dbService;
+
+    public NotificationManager(DBService dbService) {
+
+        this.dbService = dbService;
+        String notificationServiceFactoryClass = System.getProperty(ZMSConsts.ZMS_PROP_NOTIFICATION_SERVICE_FACTORY_CLASS);
```

Comment:
1) make this final

2) notification manager is not a required attribute thus notificationServiceFactoryClass can be null. So therefore we need to handle this correctly where we're running without any notification support.